### PR TITLE
Add feature flag for Multi-Subnet Cluster.

### PR DESCRIPTION
**Pull request by sawsa307**
/assign @swetharepakula 
/cc @mmamczur

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -106,7 +106,10 @@ type LoadBalancerController struct {
 
 	ZoneGetter *zonegetter.ZoneGetter
 
-	logger klog.Logger
+	// Whether the Ingress controller should process Multi-Subnet Cluster
+	// related changes
+	enableIngressMultiSubnetCluster bool
```
Comment (swetharepakula) [inline pkg/controller/controller.go:111]: rename to enableMultiSubnetCluster

We are already inside the ingress controller so don't need to add Ingress to the name

File: pkg/flags/flags.go [hunk 2]
```diff
@@ -130,6 +130,8 @@ var (
 		EnableIngressGlobalExternal              bool
 		OverrideComputeAPIEndpoint               string
 		EnableIGMultiSubnetCluster               bool
+		EnableIngressMultiSubnetCluster          bool
+		EnableNEGMultiSubnetCluster              bool
```
Comment (swetharepakula) [inline pkg/flags/flags.go:134]: do we need two? Can we do an EnableMultiSubnetCluster that either names on all clusters or disables?

cc/ @mmamczur 

Nit: can we break out the high level flag creation into its own commit

File: pkg/neg/controller.go [hunk 3]
```diff
@@ -106,6 +106,10 @@ type Controller struct {
 	// gce-regional-external ingresses
 	enableIngressRegionalExternal bool
 
+	// Whether the NEG controller should process Multi-Subnet Cluster related
+	// changes
+	enableNEGMultiSubnetCluster bool
```
Comment (swetharepakula) [inline pkg/neg/controller.go:111]: same comment, this should just be enableMultiSubnetCluster

**Comment by sawsa307** [comment]
Updated, PTAL! @swetharepakula

File: pkg/flags/flags.go (hunk 2 above)
Comment (sawsa307) [reply pkg/flags/flags.go]: Currently we have a separate flag in IG controller. I'll leave the cleanup to @mmamczur

**Comment by mmamczur** [comment]
for me, one shared flag works well. However, I remember we wanted  to have a split, specifically for the NEG controller - so I guess you decide. 
I'm ok using this shared flag in my changes.

**Comment by swetharepakula** [comment]
Even when we split, the flag will operate the same. The split just means we can run the neg controller by itself or combined.

File: pkg/flags/flags.go [hunk 4]
```diff
@@ -308,6 +309,7 @@ L7 load balancing. CSV values accepted. Example: -node-port-ranges=80,8080,400-5
 	flag.BoolVar(&F.EnableIngressGlobalExternal, "enable-ingress-global-external", true, "Enable L7 Ingress Global External. Should be disabled when Regional External is enabled.")
 	flag.StringVar(&F.OverrideComputeAPIEndpoint, "override-compute-api-endpoint", "", "Override endpoint that is used to communicate to GCP compute APIs.")
 	flag.BoolVar(&F.EnableIGMultiSubnetCluster, "enable-ig-multi-subnet-cluster", false, "Enable Multi Subnet support for the controllers that use Instance Group backends.")
+	flag.BoolVar(&F.EnableMultiSubnetCluster, "enable-multi-subnet-cluster", false, "Enable Multi Subnet support for all controllers(Ingress/NEG/L4/IG).")
```
[the remark this answers is not in the document]
Comment (sawsa307) [reply pkg/flags/flags.go]: I think we can scope this PR to only introduce flag. I'll add it to controllers when it is needed, given we have move most logic into zoneGetter

**Pull request merged**